### ISSUE TITLE
`dim` argument for `DDP` is not documented

### ISSUE THREAD

**Issue by awgu** [pytorch/pytorch#74970]
### 📚 The doc issue

https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel

The `DistributedDataParallel` constructor takes in an argument `dim=0` that is not documented. It seems to be used for scatter and gather, which are also not documented.

### Suggest a potential alternative/fix

Add some explanation for `dim`, `scatter`, and `gather`.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @brianjo @mruberry
